src/StreamJsonRpc.Tests/JsonRpcTests.cs

```diff
@@ -559,6 +559,51 @@ await invokeTask.NoThrowAwaitable(); // may or may not throw due to cancellation
         await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>(nameof(Server.TestInvalidMethod)));
     }
 
+    [Fact]
+    public async Task InvokeAsync_InvokesMethodWithAttributeSet()
+    {
+        string correctCasingResult = await this.clientRpc.InvokeWithParameterObjectAsync<string>("test/InvokeTestMethod");
+        Assert.Equal("test method attribute", correctCasingResult);
+
+        await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>("teST/InvokeTestmeTHod"));
+
+        string baseMethodResult = await this.clientRpc.InvokeAsync<string>("base/InvokeMethodWithAttribute");
+        Assert.Equal("base InvokeMethodWithAttribute", baseMethodResult);
+
+        await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>("base/InvokeVirtualMethodOverride"));
+
+        await Assert.ThrowsAsync<RemoteMethodNotFoundException>(() => this.clientRpc.InvokeAsync<string>("child/InvokeVirtualMethodOverride"));
```

Comment:
Can you add comments justifying why you expect this to fail? The child method appears to explicitly claim this is the method name, so I don't know why it wouldn't be found except perhaps that it conflicts with the base method.